BloodyMary/Navigation/Routable.swift

```diff
@@ -0,0 +1,54 @@
+//
+//  BloodyMary
+//
+//  Copyright © TheInkedEngineer. All rights reserved.
+// 
+
+import Foundation
+
+/// The Identifier to use for a `Routable`.
+public typealias ScreenIdentifier = String
+/// A navigation route.
+public typealias Route = [ScreenIdentifier]
+/// A view controller that is routable.
+public typealias RoutableViewController = Routable & UIViewController
+/// Closure called when a full navigation is completed.
+public typealias RoutingCompletion = () -> ()
+
+/// A `protocol` that all navigable `ViewControllers` should adhere to.
+/// The `UINavigationController` and the `UITabBarController` should **NOT** adhere to this protocol.
+public protocol Routable: AnyObject {
+  /// The identifier associated to the routable element.
+  /// ```swift
+  ///    extension ProfileViewController: Routable {
+  ///
+  ///      var screenIdentifier: ScreenIdentifier {
+  ///        "profile"
+  ///      }
+  ///    }
+  /// ```
+  static var screenIdentifier: ScreenIdentifier { get }
```

Comment:
You could make this available though an extension with default value:

```
String(describing: Self.self)
```

and change it wherever you want something different

Reply:
I know but the idea was to force creating it so people end up creating and enum since it is necessary to pass the dictionary of [ScreenIdentifier: VC] if I give it a default value it will be overlooked.